### ISSUE TITLE
Error when trying to add plugin 1.9.1 for iOS - undefined method `map' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by leschirmeur** [phonegap/phonegap-plugin-push#1440]
### Expected Behaviour
Plugin should be added for platform iOS without errors.

### Actual Behaviour
When trying to add the plugin version 1.9.1 for platform iOS, an error is thrown.

I have just upgraded the plugin version from 1.8.4 to 1.9.1. As stated in the docs, I've installed cocoapods using:
```
sudo gem install cocoapods
pod setup
```

When running ``` cordova plugin add phonegap-plugin-push@1.9.1 --variable SENDER_ID=<XXX> --save ``` I am receiving the error stated below in the logs.

#### Cordova CLI version and cordova platform version
cordova: 6.4.0
cordova-android: 6.1.0
cordova-ios: 4.3.1

#### Plugin version
1.9.1

#### Logs taken while reproducing problem
```
$ cordova plugin add phonegap-plugin-push@1.9.1 --variable SENDER_ID=684906563747 --save
Fetching plugin "phonegap-plugin-push@1.9.1" via npm
Installing "phonegap-plugin-push" for android
ANDROID_HOME=/Users/schirmeur/Library/Android/sdk
JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_45.jdk/Contents/Home
Subproject Path: CordovaLib
null
org.xwalk:xwalk_core_library:22+
Incremental java compilation is an incubating feature.
:clean
:CordovaLib:clean

BUILD SUCCESSFUL

Total time: 2.473 secs
Subproject Path: CordovaLib
Installing "phonegap-plugin-push" for ios
/usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/sidekick.rb:83:in `parse_results'
: undefined method `map' for nil:NilClass (NoMethodError)

Did you mean?  tap
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/sidekick.rb:32:in `search'
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/inspector.rb:65:in `search_query'
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/inspector.rb:59:in `search_exception'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/user_interface/error_report.rb:119:in `search_for_exceptions'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:65:in `report_error'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:396:in `handle_exception'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:337:in `rescue in run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:325:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:50:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/bin/pod:55:in `<top (required)>'
	from /usr/local/bin/pod:23:in `load'
	from /usr/local/bin/pod:23:in `<main>'

Failed to install 'phonegap-plugin-push':Error: pod: Command failed with exit code 1 Error output:
/usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/sidekick.rb:83:in `parse_results': undefined method `map' for nil:NilClass (NoMethodError)
Did you mean?  tap
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/sidekick.rb:32:in `search'
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/inspector.rb:65:in `search_query'
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/inspector.rb:59:in `search_exception'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/user_interface/error_report.rb:119:in `search_for_exceptions'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:65:in `report_error'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:396:in `handle_exception'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:337:in `rescue in run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:325:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:50:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/bin/pod:55:in `<top (required)>'
	from /usr/local/bin/pod:23:in `load'
	from /usr/local/bin/pod:23:in `<main>'
    at ChildProcess.whenDone (/Users/schirmeur/sites/quiply/ui/platforms/ios/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:829:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:486:12)
Error: pod: Command failed with exit code 1 Error output:
/usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/sidekick.rb:83:in `parse_results': undefined method `map' for nil:NilClass (NoMethodError)
Did you mean?  tap
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/sidekick.rb:32:in `search'
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/inspector.rb:65:in `search_query'
	from /usr/local/lib/ruby/gems/2.3.0/gems/gh_inspector-1.0.2/lib/gh_inspector/inspector.rb:59:in `search_exception'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/user_interface/error_report.rb:119:in `search_for_exceptions'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:65:in `report_error'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:396:in `handle_exception'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:337:in `rescue in run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:325:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:50:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/cocoapods-1.1.1/bin/pod:55:in `<top (required)>'
	from /usr/local/bin/pod:23:in `load'
	from /usr/local/bin/pod:23:in `<main>'
```

**Comment by macdonst** [comment]
@leschirmeur what version of node do you have installed? I'm thinking it might be old and not support the `map` function.

**Comment by leschirmeur** [comment]
Node 4.6.0
Npm 3.10.8

**Comment by leschirmeur** [comment]
Sorry for not having provided this info upfront

**Comment by macdonst** [comment]
I have a feeling your version of node does not support Array.prototype.map.
Can you try a later version of node?

**Comment by leschirmeur** [comment]
I upgraded to node 6.9.2 and npm 3.10.9 and the error still persists

**Comment by leschirmeur** [comment]
Will further investigate but if anyone might have an idea...

**Comment by leschirmeur** [comment]
From the stacktrace and by looking into the code, it seems to me that an exception is thrown here:

```from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:325:in `run'```

The affected line in code is ```plugin_prefixes.each do |plugin_prefix|```

I have honestly no idea what else I can try, reverting back to plugin version 1.8.x for the time being.

**Comment by fredgalvao** [comment]
I have a feeling that issue has nothing to do with the plugin itself, but with cordova instead. Are you able to add/remove any other plugin successfully?

**Comment by leschirmeur** [comment]
Yes, everything else works perfectly fine, it's just the phonegap-plugin-push for iOS

**Comment by coreyroth** [comment]
I've got the same issue.  With cordova-ios 4.1.1 and a blank app I can add the plugin.  With 4.3.1, I get the same error as above.  I also tried updating my version of node as it was rather dated, but it didn't change anything.  Other plugins install fine.

**Comment by macdonst** [comment]
@leschirmeur if you create a new app and add the push plugin does it work? Possibly you may need to do a:

```
cordova platform rm ios
cordova platform add ios
```

to your existing project.

**Comment by leschirmeur** [comment]
@macdonst Thanks very much for your hints, I tried installing the plugin in a new project with success. Afterwards I tried again in the project where the error was occuring and magically, it worked! So, no issue anymore.

The explanation I have is that we internally just switched from using npm to yarn for installing global packages. In this context, we removed all globally installed npm packages and added them again using yarn. So, my assumption is that some package was not properly installed, it might even have been cordova.

Now, I am running into another issue later in the build process, but this has nothing to do with this issue. Will try to solve it.

Thanks for your help!
